block-layer/codecs/DAG-JSON.md

```diff
@@ -14,14 +14,56 @@ Contrary to popular belief, JSON as a format supports Big Integers. It's only
 JavaScript itself that has trouble with them. This means JS implementations
 of `dag-json` can't use the native JSON parser and serializer.
 
+### Version 0
+
+This is an old version of `dag-json` that reserved the `"/"` key in order to
+represent binary and link data types.
+
 #### Binary Type
 
 ```javascript
 {"/": { "base64": String }}
 ```
 
-### Link Type
+#### Link Type
 
 ```javascript
 {"/": String /* base encoded CID */}
-```
\ No newline at end of file
+```
+
+### Version 1
+
+#### Format
+
+The internal data format is valid JSON but is **NOT** identical to the decoded
+node data codecs produce.
```

Comment:
I do not understand why this is necessary. Is there a place with the rationale to make JSON not JSON?

Reply:
Right now, we’re using a reserved key, `”/“`, for encoding links and binary. Several people think that key reservation is highly problematic because it means there are certain data sets you just can encode.

This change would fix that changing the internal block format. The format would **still* be valid JSON, it just wouldn’t have as close of a 1-to-1 match with the decoded form of the data.

However, this change may not be worth it. This PR is to show that we have a solution to the reserved key issue and that is what it would take to fix it.